File: nim/high-scores/high_scores.nim
import algorithm

proc latest*(scores: seq[int]): int =
  scores[^1]

proc personalBest*(scores: seq[int]): int =
  scores.max

proc personalTopThree*(scores: seq[int]): seq[int] =
  scores.sorted.reversed[0 ..< 3.min(scores.len)]

File: r/high-scores/high-scores.R
scores_list <- function(scores) {
  scores
}

latest <- function(scores) {
  tail(scores, n = 1)
}

personal_best <- function(scores) {
  max(scores)
}

personal_top_three <- function(scores) {
  sort(scores, decreasing = TRUE)[1:(min(length(scores), 3))]
}
